xarray/core/duck_array_ops.py

```diff
@@ -279,14 +279,51 @@ def f(values, axis=None, skipna=None, **kwargs):
 _mean = _create_nan_agg_method('mean')
 
 
+def datetime_to_numeric(array, offset=None, datetime_unit=None, dtype=float):
+    """Convert an array containing datetime-like data to an array of floats.
+
+    Parameters
+    ----------
+    da : array
+        Input data
+    offset: Scalar with the same type of array or None
+        If None, subtract minimum values to reduce round off error
+    datetime_unit: None or any of {'Y', 'M', 'W', 'D', 'h', 'm', 's', 'ms',
+        'us', 'ns', 'ps', 'fs', 'as'}
+    dtype: target dtype
+
+    Returns
+    -------
+    array
+    """
+    if offset is None:
+        offset = array.min()
+    array = array - offset
+
+    if datetime_unit:
+        array = array / np.timedelta64(1, datetime_unit)
```

Comment:
I think we need to move this to after we potentially convert an array of `datetime.timedelta` objects to `np.timedelta64` (otherwise this will raise an error).

Reply:
To explain this a little more -- for cftime dates, `array - offset` (a few lines above) will return an array of `datetime.timedelta` objects.  When `datetime_to_numeric` was used with `xarray.Variable` objects, we did not have to worry about converting to `np.timedelta64`, because that would happen automatically.  In the case of pure NumPy or dask arrays, we do.